fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java

```diff
@@ -248,7 +248,19 @@ public TListTableStatusResult listTableStatus(TGetTablesParams params) throws TE
         if (db != null) {
             db.readLock();
             try {
-                for (Table table : db.getTables()) {
+                List<Table> tables = null;
+                if (!params.isSetType() || params.getType() == null || params.getType().isEmpty()) {
+                    tables = db.getTables();
+                } else {
+                    switch (params.getType()) {
+                        case "VIEW":
+                            tables = db.getViews();
+                            break;
+                        default:
+                            tables = db.getTables();
+                    }
+                }
```

Comment:
```suggestion
               List<Table> tables = db.getTables();
                if (params.isSetType() && "VIEW".equals(params.getType())) {
                    tables = db. getViews();
                }
```

Reply:
If type is view, this will call getTables() which is not required. When the count of tables in this db is too many, this will take more time.